src/decision/location_selection.jl

```diff
@@ -0,0 +1,266 @@
+end
+function ranks_to_frequencies(
+    ranks::NamedDimsArray{D,T,2,A};
+    n_ranks=length(ranks.sites),
+    agg_func=nothing,
+) where {D,T,A}
+    if !isnothing(agg_func)
+        return agg_func(ranks_to_frequencies(ranks, n_ranks))
+    end
+
+    return ranks_to_frequencies(ranks, n_ranks)
+end
+
+"""
+    location_selection_frequencies(ranks::NamedDimsArray;n_loc_int::Int64=5)
+    location_selection_frequencies(inv_log::NamedDimsArray{D,T,4,A};dims::Union{Symbol,Vector{Symbol}}=:coral_id) where {D,T,A}
+
+Post-processing function for intervention logs. Calculates the frequencies with which locations were selected for a particular intervention,
+for a selection of scenarios (e.g. selected robust scenarios).
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `n_loc_int` : number of locations which are intervened at for each intervention decision.
+- `dims` : dimensions to sum selection frequencies over.
+
+# Returns 
+Counts for location selection at each location in the domain.
+"""
+function location_selection_frequencies(
+    ranks::NamedDimsArray;
+    n_loc_int::Int64=5,
+)
+    ranks_frequencies = ranks_to_frequencies(ranks; n_ranks=n_loc_int)
+    loc_count = sum(ranks_frequencies[ranks=1:n_loc_int], dims=2)[ranks=1]
+
+    return loc_count
+end
+function location_selection_frequencies(
+    inv_log::NamedDimsArray{D,T,4,A};
+    dims::Union{Symbol,Vector{Symbol}}=:coral_id,
+) where {D,T,A}
+    loc_count = dropdims(
+        sum(dropdims(sum(inv_log; dims=dims); dims=dims) .> 0; dims=:scenarios);
+        dims=:scenarios,
+    )
+    return loc_count
+end
+
+"""
+    summed_inverse_rank(ranks::NamedDimsArray{D,T,3,A};dims::Union{Symbol,Vector{Symbol}}=[:scenarios, :timsteps],
+            ) where {D,T,A}
+    summed_inverse_rank(ranks::NamedDimsArray{D,T,2,A};) where {D,T,A}
+    summed_inverse_rank(ranks::NamedDimsArray,dims::Union{Symbol,Vector{Symbol}},)
+
+Calculates (number of sites) .- ranks summed over the dimension dims and transformed using agg_func 
+    (default no transformation).
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `dims` : Dimensions to sum over.
+
+# Returns 
+Inverse rankings (i.e. the greater the number the higher ranked the site).
+"""
+function summed_inverse_rank(
+    ranks::NamedDimsArray{D,T,3,A};
+    dims::Vector{Symbol}=[:scenarios, :timesteps],
+) where {D,T,A}
+    return summed_inverse_rank(ranks, dims)
+end
+function summed_inverse_rank(
+    ranks::NamedDimsArray{D,T,2,A};
+    dims::Vector{Symbol}=[:scenarios],
+) where {D,T,A}
+    return summed_inverse_rank(ranks, dims)
+end
+function summed_inverse_rank(
+    ranks::NamedDimsArray,
+    dims::Vector{Symbol},
+)
+    n_ranks = maximum(ranks)
+    inv_ranks = dropdims(sum(n_ranks .- ranks; dims=dims); dims=dims[1])
```

Comment:
Since this function receives a NamedDimsArray as input and doesn't give support for simples Arrays, why not pass the dimension you want to drop as a key (I assume is `:timestamp`, right?) instead of a number?

Reply:
The dimension here is a key, it's just when using dropdims you don't want to drop all the dimensions that are summed over, just the first one

Reply:
Oh, ok, my bad!